src/datasources/errors/validation-error-factory.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
+import { DefinedError } from 'ajv';
+import { HttpExceptionPayload } from './interfaces/http-exception-payload.interface';
+
+/**
+ * Creates an {@link HttpException} from an array of validation errors.
+ * Both http status and internal error code are fixed.
+ * Details about the validation error are logged but hidden from the client response.
+ */
+@Injectable()
+export class ValidationErrorFactory {
+  private readonly logger = new Logger(ValidationErrorFactory.name);
+  private readonly VALIDATION_ERROR_CODE = 42;
+
+  from(errors: DefinedError[]): HttpException {
+    const errPayload: HttpExceptionPayload = {
+      message: 'Validation failed',
+      code: this.VALIDATION_ERROR_CODE,
+      arguments: [],
+    };
+
+    const detail = errors.map((err) => JSON.stringify(err));
+    this.logger.error({ ...errPayload, detail });
```

Comment:
I believe that `JSON.stringify` can be quite an expensive operation if we have a complex/sizeable object. We are using this just to log the error. Do you have a use-case that I might be missing if we don't `stringify` it?

Reply:
Fixed in [dc0377f](https://github.com/5afe/safe-client-gateway-nest/commit/dc0377fa957d443f10ee5d31c65f3da203e40628)